### PR TITLE
Relayout table view cell nodes if there is a mismatch between content view size and the node's constrained size

### PR DESCRIPTION
**Pull request by nguyenhuy**
- Above is the generic case. Correctly handling it means relayout when the table view enters or leaves editing mode is taken care of as well.
- Async data source API removal: In a table view, cell nodes should always fill its content view and table view widths. Thus async data source can no longer provide custom constrained size for cell nodes. This removal allows table view to better handle relayout.
- Some more tests are added to ASTableViewTests to check against use cases handled in this diff.

### CODE REVIEW

**Comment by nguyenhuy** [comment]
With this, I'm confident that #512 is finally solved.

File: AsyncDisplayKit/ASTableView.mm [hunk 1]
```diff
@@ -104,22 +105,30 @@ - (id)forwardingTargetForSelector:(SEL)aSelector
...
+- (void)layoutSubviews
+{
+  [_delegate willLayoutSubviewsOfTableViewCell:self];
+  [super layoutSubviews];
+}
+
 - (void)didTransitionToState:(UITableViewCellStateMask)state
 {
+  [self setNeedsLayout];
   [super didTransitionToState:state];
```
Comment (appleguy) [inline AsyncDisplayKit/ASTableView.mm]: I wonder, should we actually be calling didTransitionToState: before the layout has been completed?  Maybe we should add a call to -layoutIfNeeded here.

File: AsyncDisplayKitTests/ASTableViewTests.m [hunk 2]
```diff
@@ -290,4 +294,138 @@ - (void)triggerSizeChangeAndAssertRelayoutAllRowsForTableView:(ASTableView *)tab
   }];
 }
 
+- (void)testRelayoutVisibleRowsWhenEditingModeIsChanged
```
Comment (appleguy) [inline AsyncDisplayKitTests/ASTableViewTests.m]: Love the new tests — thanks!  Code coverage in ASDK continues to increase, and I think more of it is due to your tests than anyone else.

File: AsyncDisplayKit/ASTableView.mm (hunk 1 above)
Comment (nguyenhuy) [inline AsyncDisplayKit/ASTableView.mm]: Done.

**Pull request merged**